class-search-swagger.yaml

```diff
@@ -0,0 +1,250 @@
+      operationId: getClasses
+      # This is array of GET operation parameters:
+      produces:
+        - application/json
+      parameters:
+        - in: query
+          name: term
+          description: "The term code is based on the fiscal year. To find the term that you an appropriate value, first query `/terms/open` or `/terms` and use the `code` value."
+          required: true
+          type: string
+        - in: query
+          name: subject
+          description: "This is a two to four letter course subject. To find a list of appropriate subjects, you can query: `/course-subjects` and use the `abbreviation` value."
+          required: false
+          type: string
+        - in: query
+          name: courseNumber
+          description: "This is the course number associated with a class. For example, the value `111` is the course number for MTH 111."
+          required: false
+          type: string
+        - in: query
+          name: q
+          description: "Searches class title, description, and subject"
+          required: false
+          type: string
+        - in: query
+          name: page[size]
+          description: "Number of results to return. Used in pagination. Default value is 10."
+          required: false
+          type: string
+        - in: query
+          name: page[number]
+          description: "Page number of results. Used to paginate through results."
+          required: false
+          type: string
+        
+          
+      # Expected responses for this operation:
+      responses:
+        # Response code
+        "200":
+          description: "Successful response"
+          schema:
+            $ref: "#/definitions/ResultObjects"
+        "404":
+          description: "Not Found"
+          schema:
+            $ref: "#/definitions/Error"
+        "500":
+          description: "Internal Server Error"
+          schema:
+            $ref: "#/definitions/Error"
+
+definitions:
+  ResultObjects:
+    properties:
+      links:
+        $ref: "#/definitions/Links"
+      data:
+        type: array
+        items:
+          $ref: "#/definitions/ResourceObject"
+        
+  ResourceObject:
+    properties:
+      id:
+        type: string
+      type:
+        type: string
+      attributes:
+        $ref: "#/definitions/Attributes"
+        
+  Attributes:
+    properties:
+      campusDescription:
+        type: string
+        description: "Name of campus. Sample value: `Oregon State - Corvallis`"
+      courseNumber:
+        type: string
+        description: "Number of course. Sample value: `111` for MTH 111"
+      crn:
+        type: string
+        description: "Course Reference Number used to uniquely identify a class available for registration. This is the CRN value that students use to register for a class section."
+      sectionTitle:
+        type: string
+        description: "Title of the course section"
+      creditHourHigh:
+        type: string
```

Comment:
type: integer?

Same for creditHourLow, creditHours, enrollment, maximumEnrollment, waitCapacity, waitCount.